src/main/kotlin/org/partiql/scribe/problems/ScribeProblem.kt

```diff
@@ -0,0 +1,93 @@
+package org.partiql.scribe.problems
+
+import org.partiql.spi.Enum
+import org.partiql.spi.UnsupportedCodeException
+
+/**
+ * Represents a problem that is encountered during transpilation.
+ *
+ * @property code
+ * @property severity The severity of the problem.
+ * @property properties The properties of the problem.
+ */
+public class ScribeProblem(
```

Comment:
self-review: vend a Scribe-specific error API that is based off of PLK's [PError](https://github.com/partiql/partiql-lang-kotlin/blob/main/partiql-spi/src/main/java/org/partiql/spi/errors/PError.java).

Goals of this error API:
- allow future public API additions without breaking customers
- allow for users of API to extract relevant error codes and properties from the errors
- allow for collecting multiple erorrs/warnings

PError provides for all of the above but doesn't provide extension points to add more Scribe errors hence why there are some similar looking classes in Scribe.